source/game.programme.newsevent.bmx

```diff
@@ -502,33 +504,66 @@ Type TNewsEvent Extends TBroadcastMaterialSource {_exposeToLua="selected"}
 	End Method
 	
 	
-	Method ReplacePlaceholdersFromTemplate:Int(template:TNewsEventTemplate = Null, time:Long = 0)
+	Method ReplacePlaceholdersFromTemplate:Int(template:TNewsEventTemplate = Null, parentVariables:TTemplateVariables = null, time:Long = 0)
 		If not template and not self.templateID Then Return False
 		If not template
 			template = GetNewsEventTemplateCollection().GetByID(self.templateID)
 			If not template Then Return False
 		EndIf
 
+		Local varToUse:TTemplateVariables
+		If templateVariables
+			varToUse = templateVariables
+		Else
+			varToUse = parentVariables
+			If Not varToUse
+				varToUse = template.templateVariables
+			ElseIf template.templateVariables
+				'both template variables and parent variables exist - check if merging is necessary
+				Local mergeNecessary:Int = False
+				'TODO nested variables with parent are not yet considered by this check
+				For Local key:Object = EachIn parentVariables.variables.keys()
+					If Not template.templateVariables.variables.Contains(key)
+						'parent contains additional variable
+						mergeNecessary = True
+						Exit
+					EndIf
+				Next
+				If mergeNecessary
+					throw "implement merge of template variables - cause: " + template.GetTitle()
+				Else
+					'for self triggered news merge will not be necessary - parent variables are ignored
+				EndIf
+			EndIf
+		EndIf
+
 		'copy text if we intend to replace content
 		'(for now only check main language)
 		If template.title.Get().Find("%") >= 0 or template.title.Get().Find("${") >= 0
-			If template.templateVariables
-				Self.title = _ReplacePlaceholders(template.templateVariables.ReplacePlaceholders(template.title), time)
+			If varToUse
+				Self.title = _ReplacePlaceholders(varToUse.ReplacePlaceholders(template.title), time)
 			Else
 				Self.title = _ReplacePlaceholders(template.title, time)
 			EndIf
 		Else
 			Self.title = template.title
 		EndIf
 		If template.description.Get().Find("%") >= 0 or template.description.Get().Find("${") >= 0
-			If template.templateVariables
-				Self.description = _ReplacePlaceholders(template.templateVariables.ReplacePlaceholders(template.description), time)
+			If varToUse
+				Self.description = _ReplacePlaceholders(varToUse.ReplacePlaceholders(template.description), time)
 			Else
 				Self.description = _ReplacePlaceholders(template.description, time)
 			EndIf
 		Else
 			Self.description = template.description
 		EndIf
+
+		'store variables for passing on to potential trigger
+		'TODO store "copy" rather than original
```

Comment:
Willst du das noch angehen?

Reply:
Tatsächlich nicht für diesen PR. Ich glaube, das wird relevanter für den weiter oben genannten Fall.
Wird die Initialnachricht gesendet, werden alle Variablen (dieser Nachricht) zurückgesetzt. So lange derselbe Nachrichtenthread also nicht gestartet wird, bis der vorige beendet ist und alle Variable in der Hauptnachricht definiert sind, gibt es kein Problem.
Erst wenn man eigene "Untervariablen" in getriggerten Nachrichten erlauben will, muss man sich überlegen, wie man diese mit den Parentvariablen zusammenbringt.

Reply:
Den gleichen Thread startet man ja nicht nochmal. Was passieren kann ist, dass aus der gleichen "Vorlage" (Template) eine neue Kette an Ereignissen generiert wird.
Sollte also kein Problem dahingehend geben.

Reply:
Dann könnte es aber schon ein Problem sein, denn `news = New TNewsEvent.InitFromTemplate(template, variables)` übergibt ja die originalen Templatevariablen, keinen Klon. Schöne wäre ein variables.copy(), das eine neue Instanz der Variablen erzeugt. Die könnte man dann sicher dem neuen "Newsthread" mitgeben.

Reply:
Ich habe das Ticket aufgemacht und Dir zugewiesen. Im besten Fall baust Du das Copy und ich verwende es dann hier gleich.